### ISSUE TITLE
prune.rpart getting subscript out of bounds

### ISSUE THREAD

**Issue by davavra** [bethatkinson/rpart#4]
prune.rpart gets a subscript out of bounds error on the last line shown here
`    temp <- pmax(tree$cptable[, 1L], cp)
    keep <- match(unique(temp), temp)
    newx$cptable <- tree$cptable[keep, , drop = FALSE]
    newx$cptable[max(keep), 1L] <- cp
`
The values in keep may exceed the number of rows in newx$cptable. In fact, this is causing the error I've encountered where max(keep)=298 and the number of rows in  newx$cptable is 295. The line probably should be altered to read:
`    newx$cptable[length(keep), 1L] <- cp
`
but I don't know the purpose of this statement so I'm not sure this would be the proper fix.

**Comment by bethatkinson** [comment]
do you have a reproducible example that I can work with?

**Comment by davavra** [comment]
Yes, but it's huge and proprietary.I  don't have a simpler example.
DAV
